Fix SVN dirty calculation

sodevel opened the pull request:
I discovered that my svn working copies always get marked as dirty although they have been clean. The issue is that the logic of the dirty calculation is broken and detects externals always as dirty. While fixing this, i also discovered that the dirty state of the root directory of the working copy was not detected. This is because the relative path calculation returns an empty path for the root directory which is always marked as clean.

I don't know why the entry `/userHome` or root should always be marked as clean so i simply removed this calculation and test.

dcoraboeuf commented:
Hi @sodevel , first of all, many thanks for this contribution and I'm sorry for the delay (for any reason, I did not receive any notification any longer about this repository, which was very unfortunate, and I did not see your PR).

Now, I have merged some previous PRs and your code is now in conflict with some other changes.

Would you be so kind as porting your changes to the new code by solving these conflicts against the `release/2.14` branch?

Thanks,
Damien

sodevel commented:
I rebased against the new base branch and fixed the conflicts. The code is looking good, however currently i can't access my usual environment and therefor can't build and test the plugin to verify.

sodevel commented:
Just a small update, now i could build and test the plugin, it is working for me.
